Tes 71 filter component

KozerT opened the pull request:
Create responsive Filter component with mock data. 

Tests will be implemented once the Context and Fetch functionality are finalized, along with decisions on how the fetched data will be displayed.
<img width="375" alt="Screenshot 2024-09-30 at 14 56 27" src="https://github.com/user-attachments/assets/eead00d8-4cd5-4180-830d-d91242b0484b">

<img width="620" alt="Screenshot 2024-09-30 at 14 56 19" src="https://github.com/user-attachments/assets/eebfb813-0a6d-4198-9254-3f11d63c7fb2">

<img width="1212" alt="Screenshot 2024-09-30 at 14 54 14" src="https://github.com/user-attachments/assets/2824ec64-5740-4c75-9b7a-7241f68987a3">

<img width="637" alt="Screenshot 2024-09-30 at 14 56 03" src="https://github.com/user-attachments/assets/5cff92f4-fa72-48b6-a024-c9102a2e1f3b">

KozerT commented on src/components/ui/modal.tsx:
Removed this, since we don't use this anywhere.

WitchDevelops commented on src/components/filters/CheckBoxLargeScrn.tsx:
```diff
@@ -0,0 +1,52 @@
+                    key={option}
+                    className='flex flex-row items-center space-x-2 align-middle text-base'
+                >
+                    <Checkbox
```
It's a very well-structured component that renders checkbox groups on large devices, it follows best practices and is very readable.

🟡 I suggest either wrapping the <Checkbox> in the <label> tag or adding an id prop to the checkbox (or both), to improve the UX. This way clicking on the label checks the associated checkbox and adding an id prop to the checkbox.

like this:
```jsx
<label htmlFor={option} className='flex items-center space-x-2 hover:cursor-pointer'>
  <Checkbox
    id={option}
    checked={selectedValues.includes(option)}
    onCheckedChange={(checked: boolean) => {
    handleCheckboxChange(option, checked);
    }}
   />
  <span className='font-light'>{option}</span> //span element optional
</label>
```

WitchDevelops commented on src/components/modals/FilterModal.tsx:
```diff
@@ -0,0 +1,155 @@
+    priceRanges: z.array(z.string()).optional(),
+    colors: z.array(z.string()).optional(),
+    stock: z.array(z.string()).optional(),
+});
```
since the schema for each item is the same, maybe it could be abstracted to some constant or type and reused to limit duplication?

WitchDevelops commented on src/components/modals/FilterModal.tsx:
```diff
@@ -0,0 +1,155 @@
+            </DialogTrigger>
+            <DialogContent className='flex max-h-screen flex-col overflow-y-auto p-4 [&>button]:hidden'>
+                <DialogTitle>Filter By</DialogTitle>
+                <DialogDescription className='sr-only'>
```
Great catch for including screen readers!

WitchDevelops commented on src/types/global.d.ts:
in the schema Henrika is using on the backend there are more brands: 
```sql
INSERT INTO brands (id, name)
VALUES (1, 'Samsung'), (2, 'Xiaomi'), (3, 'Apple'), (4, 'POCO'), (5, 'Nokia'), (6, 'Sony'),
(7, 'Redmi'), (8, 'LG');
```

there are more colors:
```sql
INSERT INTO colors (id, name)
VALUES (1,'Black'), (2,'Blue'), (3,'White'), (4,'Green'), (5,'Purple'), (6,'Grey'), (7,'Yellow'),
(8,'Silver'), (9,'Pink'), (10,'Red'), (11,'Almond'), (12,'Lavender'), (13,'Orange');
```

price ranges are different:

```sql
INSERT INTO price_intervals (id, name, min_price, max_price)
VALUES (1, 'price_monthly_0_10', 0.00, 10.00), (2, 'price_monthly_10_50', 10.00, 50.00),
(3, 'price_monthly_50_100', 50.00, 100.00),(4, 'price_monthly_100_150', 100.00, 150.00),
(5, 'price_monthly_150_200', 150.00, 200.00);
```

WitchDevelops reviewed (changes requested):
Great job! Your code is well structured and type safe and adds a required feature. naming convention is clear.

🔴there are issues with the labels that need to be addressed before merging:
- on mobile devices: clicking on the label checks/unchecks the first checkbox in the list. this is not desired. clicking on the label should check/uncheck associated checkbox.
- on desktop: clicking on the label does nothing to the associated checkbox (it should check/uncheck it)

KozerT commented:
I've fixed the label issues, implemented the recommended changes, and added the formatPriceRange function in the utils (used in FilterModal and Filters , where data will be fetched) .

For now, this PR focuses on the frontend updates. I could link this to the backend in a separate PR 


<img width="461" alt="Screenshot 2024-09-30 at 23 57 27" src="https://github.com/user-attachments/assets/e19aad58-b54a-4929-967c-ccab1be64cee">
